R/get_stan_data.R

```diff
@@ -42,68 +42,230 @@ get_input_ww_data_for_stan <- function(preprocessed_ww_data,
-    lod_col_name = "log_lod"
+  return(ww_data)
+}
+
+#' Get date time spine to map to model output
+#'
+#' @param forecast_date a character string in ISO8601 format (YYYY-MM-DD)
+#' indicating the date that the forecast is to be made.
+#' @param input_count_data a dataframe of the count data to be passed
+#' directly to stan, , must have the following columns: date, count, total_pop
+#' @param last_count_data_date string indicating the date of the last observed
+#' count data point in 1SO8601 format (YYYY-MM-DD)
+#' @param calibration_time integer indicating the number of days to calibrate
+#' the model for, default is `90`
+#' @param forecast_horizon integer indicating the number of days, including the
+#' forecast date, to produce forecasts for, default is `28`
+#'
+#'
+#' @return a tibble containing an integer for time mapped to the corresponding
+#' date, for the entire calibration and forecast period
+#' @export
+#'
+get_date_time_spine <- function(forecast_date,
+                                input_count_data,
+                                last_count_data_date,
+                                calibration_time,
+                                forecast_horizon) {
+  nowcast_time <- as.integer(
+    lubridate::ymd(forecast_date) - last_count_data_date
   )
+  date_time_spine <- tibble::tibble(
+    date = seq(
+      from = min(input_count_data$date),
+      to = min(input_count_data$date) +
+        calibration_time +
+        nowcast_time +
+        forecast_horizon,
+      by = "days"
+    )
+  ) |>
+    dplyr::mutate(t = row_number())
+  return(date_time_spine)
+}
 
-  # Filter out wastewater outliers, and remove extra wastewater
-  # data. Arrange data for indexing. This is what will be returned.
-  ww_data <- preprocessed_ww_data |>
-    dplyr::filter(
-      .data$exclude != 1,
-      .data$date > !!last_count_data_date -
-        lubridate::days(!!calibration_time)
-    ) |>
-    dplyr::arrange(.data$date, .data$lab_site_index)
+#' Get mapping from lab-site to site
+#'
+#' @param input_ww_data a dataframe of the wastewater data to be passed
+#' directly to stan, must have the following columns: date, site, lab,
+#' genome_copies_per_ml, site_pop, below_lod, and exclude
+#'
+#' @return a dataframe mapping the unique combinations of sites and labs
+#' to their indices in the model and the population of the site in that
+#' observation unit (lab_site)
+#' @export
+#'
+get_lab_site_site_spine <- function(input_ww_data) {
+  ww_data_present <- !is.null(input_ww_data)
+
+  if (ww_data_present) {
+    lab_site_site_spine <-
+      input_ww_data |>
+      dplyr::select(
+        "lab_site_index", "site_index",
+        "site", "lab", "site_pop"
+      ) |>
+      dplyr::arrange(.data$lab_site_index) |>
+      dplyr::distinct() |>
+      dplyr::mutate(
+        "lab_site_name" = glue::glue(
+          "Site: {site}, Lab: {lab}"
+        )
+      )
+  } else {
+    lab_site_site_spine <- tibble::tibble()
```

Comment:
Do you want this to be an empty tibble or a zero-length tibble with the appropriate column names?

Reply:
I think I want it to be empty, since there is no mapping in this case. I don't think there is a need for it to have the correct column names.